btr-api/src/btr_api/services/validator.py

```diff
@@ -19,3 +25,78 @@ def validate_entity(entity: dict) -> list[str]:
         errors.append('The business is frozen')
 
     return errors
+
+
+def _get_todo_header(todo: dict) -> dict:
+    return todo.get('task', {}).get('todo', {}).get('header', {})
+
+
+def _validate_ar_filing(todos: list, submission: dict) -> []:
+    filing_year = submission.get('arFilingForYear', None)
+    if not filing_year:
+        return ['Missing arFilingForYear']
+
+    for todo in todos:
+        todo_enabled = todo.get('enabled', False)
+        if todo_enabled:
+            tr_filing_year = _get_todo_header(todo).get('TRFilingYear')
+            subtype = _get_todo_header(todo).get('subType')
+            if tr_filing_year is not None and tr_filing_year == filing_year and subtype == 'annual':
+                return []
+
+    return ['Invalid arFilingForYear']
+
+
+def _find_initial_filing_todo(todos: list) -> Optional[dict]:
+    for todo in todos:
+        todo_enabled = todo.get('enabled', False)
+        if todo_enabled:
+            todo_name = _get_todo_header(todo).get('name')
+            todo_sub_type = _get_todo_header(todo).get('subType')
+            if todo_name == 'tranparencyRegister' and todo_sub_type == 'initial':
+                return todo
+
+    return None
+
+
+def _validate_initial_filing(todos: list) -> []:
+    # require to have todo for initial filing
+    if _find_initial_filing_todo(todos) is not None:
+        return []
+
+    return ['Invalid filingType']
+
+
+def _validate_change_filing(todos: list) -> []:
+    # initial filing should be filed, meaning no enabled initial filing todo should be found
+    if _find_initial_filing_todo(todos) is None:
+        return []
+
+    return ['Invalid filingType']
+
+
+def validate_tr_filing_for_type(submission: dict, jwt: JwtManager):
+    """
+    Validate the filing on its type and available todos to verify if this filing type can be filed.
+
+    Parameters:
+    filing (dict): The filing data to validate
+
+    Returns:
+    list[str]: A list of errors, if any, found during validation
+    """
+    filing_type = submission['filingType']
+    business_identifier = submission['businessIdentifier']
+    entity_service = EntityService(current_app)
+    todos = entity_service.get_entity_todos(business_identifier, jwt)
```

Comment:
Adding on to my previous comment this would be:
`(entity_service.get_entity_info(jwt, {business_identifier}/tasks)).json().get('tasks', [])`